From 3b3c5e73ef810e5884a1f0ab42a5f2eddb3bfbe2 Mon Sep 17 00:00:00 2001
From: Yehuda Yitschak <yehuday@marvell.com>
Date: Tue, 28 Jul 2015 16:37:33 +0300
Subject: [PATCH 243/538] make: change TEXT_OFFSET for emulation platform

To avoid copying the image during booti command we change
the TEXT_OFFSET to match the load address in u-boot

Change-Id: Ibca88d99309470b5f6860a9a09d5dab2aa0a632c
Signed-off-by: Yehuda Yitschak <yehuday@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/22264
Tested-by: Star_Automation <star@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/28082
---
 arch/arm64/Makefile | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm64/Makefile b/arch/arm64/Makefile
index b6c90e5..cf65c159 100644
--- a/arch/arm64/Makefile
+++ b/arch/arm64/Makefile
@@ -53,8 +53,12 @@ head-y		:= arch/arm64/kernel/head.o
 ifeq ($(CONFIG_ARM64_RANDOMIZE_TEXT_OFFSET), y)
 TEXT_OFFSET := $(shell awk 'BEGIN {srand(); printf "0x%03x000\n", int(512 * rand())}')
 else
+ifeq ($(CONFIG_ARCH_MVEBU_PD), y)
+TEXT_OFFSET := 0x00180000
+else
 TEXT_OFFSET := 0x00080000
 endif
+endif
 
 # KASAN_SHADOW_OFFSET = VA_START + (1 << (VA_BITS - 3)) - (1 << 61)
 # in 32-bit arithmetic
-- 
1.9.1

